backend/azure/options.go

```diff
@@ -10,6 +16,47 @@ type Options struct {
 	// AccountKey holds the Azure Blob Storage account key for authentication
 	AccountKey string
 
+	// TenantID holds the Azure Service Account tenant id for authentication
+	TenantID string
+
+	// ClientID holds the Azure Service Account client id for authentication
+	ClientID string
+
+	// ClientSecret holds the Azure Service Account client secret for authentication
+	ClientSecret string
+
+	// AzureEnvName holds the name for the Azure environment
+	AzureEnvName string
+
 	// RetryFunc holds the retry function
 	RetryFunc vfs.Retry
+
+	tokenCredentialFactory TokenCredentialFactory
+}
+
+func NewOptions() *Options {
```

Comment:
Weird that golangci-lint isn't reporting the golint returns but we should add comments to any exported functions/types/vars etc.